node_modules/gh-series/lib/internal/patterns/cambridge/fullpattern.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2014 "Fronteer LTD"
+ * Grasshopper Event Engine
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var _ = require('lodash');
+
+var Pattern = require('./pattern').Pattern;
+
+/**
+ * The full pattern model. Given a set of dates, this model is able
+ * to generate a full "timetable serie pattern".
+ */
+var FullPattern = module.exports.FullPattern = function() {
+    var that = {
+        'patterns': []
+    };
+
+    /**
+     * Add an event that should be included in the pattern
+     *
+     * @param  {Event}      event       The event to include in the pattern
+     */
+    that.add = function(event) {
+        var pattern = new Pattern(event.start, event.end);
```

Comment:
We don't have to. Because we're exploding each pattern into `DayTime` and `TermWeek` instances which then get merged, the order of the events doesn't really matter. I've added a unit test, just in case though.